sdk/core/core-https/src/pipeline.ts

```diff
@@ -343,3 +365,100 @@ class HttpsPipeline implements Pipeline {
 export function createEmptyPipeline(): Pipeline {
   return HttpsPipeline.create();
 }
+
+/**
+ * Options that allow configuring redirect behavior.
+ */
+export interface PipelineRedirectOptions extends RedirectPolicyOptions {
+  /**
+   * If true, disables automatic following of redirects.
+   */
+  disable?: boolean;
+}
+
+/**
+ * Defines options that are used to configure the HTTP pipeline for
+ * an SDK client.
+ */
+export interface PipelineOptions {
+  /**
+   * The HttpsClient implementation to use for outgoing HTTP requests.
+   * Defaults to DefaultHttpsClient.
+   */
+  httpClient?: HttpsClient;
+
+  /**
+   * Options that control how to retry failed requests.
+   */
+  retryOptions?: ExponentialRetryPolicyOptions;
+
+  /**
+   * Options to configure a proxy for outgoing requests.
+   */
+  proxyOptions?: ProxySettings;
+
+  /**
+   * Options for how HTTP connections should be maintained for future
+   * requests.
+   */
+  keepAliveOptions?: KeepAlivePolicyOptions;
+
+  /**
+   * Options for how redirect responses are handled.
+   */
+  redirectOptions?: PipelineRedirectOptions;
+
+  /**
+   * Options for adding user agent details to outgoing requests.
+   */
+  userAgentOptions?: UserAgentPolicyOptions;
+}
+
+/**
+ * Defines options that are used to configure internal options of
+ * the HTTP pipeline for an SDK client.
+ */
+export interface InternalPipelineOptions extends PipelineOptions {
+  /**
+   * Options to configure request/response logging.
+   */
+  loggingOptions?: LogPolicyOptions;
+
+  /**
+   * Configure whether to decompress response according to Accept-Encoding header (node-fetch only)
+   */
+  decompressResponse?: boolean;
+}
+
+/**
+ * Create a new pipeline with a default set of customizable policies.
+ * @param options Options to configure a custom pipeline.
+ */
+export function createPipelineFromOptions(options: InternalPipelineOptions): Pipeline {
+  const pipeline = HttpsPipeline.create();
+
+  if (isNode) {
+    pipeline.addPolicy(proxyPolicy(options.proxyOptions));
+
+    if (options.decompressResponse === false) {
```

Comment:
nit: do we want to add `isNode` check for disableResponseCompressionPolicy similar what we have for proxyPolicy?

Reply:
I'm confused what you mean, this is inside an `isNode` block, and the policy only is applicable in Node.

Reply:
sorry I misread the blocks and thought they are different blocks at the same level